src/Microsoft.ML.PipelineInference/PurposeInference.cs

```diff
@@ -318,8 +318,10 @@ private static IEnumerable<IPurposeInferenceExpert> GetExperts()
         /// <param name="data">The data to use for inference.</param>
         /// <param name="columnIndices">Indices of columns that we're interested in.</param>
         /// <param name="args">Additional arguments to inference.</param>
+        /// <param name="columnPurpose">(Optional) User defined mapping of Column to Purpose.</param>
         /// <returns>The result includes the array of auto-detected column purposes.</returns>
-        public static InferenceResult InferPurposes(IHostEnvironment env, IDataView data, IEnumerable<int> columnIndices, Arguments args)
+        public static InferenceResult InferPurposes(IHostEnvironment env, IDataView data, IEnumerable<int> columnIndices, Arguments args,
+            Dictionary<string, ColumnPurpose> columnPurpose = null)
```

Comment:
>Dictionary<string, ColumnPurpose> columnPurpose = null [](start = 12, length = 54)

Why is this needed here? Should we intersect the result of this, with the purposes we have, in the places that call this?

Reply:
We need it here because we want to first honor the column purpose set by the user -- In line #344 we set the 'SuggestedPurpose' to be what the user provided.  By doing this we ensure the  'Experts' below do not  do purpose inference  (and hence transform inference) on those user provided columns.   Auto inference will only be done on columns  which do not  have user intent 

---
In reply to: [199592844](https://github.com/dotnet/machinelearning/pull/461#discussion_r199592844) [](ancestors = 199592844)